### PR TITLE
[1/n] [reconfigurator] add a test wrapper around realize_blueprint_with_overrides

### PR DESCRIPTION
**Pull request by sunshowers**
We're going to do more things in this wrapper in #6399.

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -3616,7 +3616,7 @@ dependencies = [
  "httpdate",
  "itoa",
  "pin-project-lite",
- "socket2 0.4.10",
```
Comment (sunshowers) [inline Cargo.lock:3619]: This has sadly been flapping between socket 0.4 and socket 0.5 for a long time :(

File: nexus/reconfigurator/execution/src/test_utils.rs [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        datastore,
+        resolver,
+        blueprint,
+        Uuid::new_v4(),
+        overrides,
+    )
+    .await
+    // We expect here rather than in the caller because we want to assert
+    // that the result is a `RealizeBlueprintOutput` (which is `must_use`
+    // so it must be assigned to `_`).
```
Comment (jgallagher) [inline nexus/reconfigurator/execution/src/test_utils.rs:32]: I'm not following this comment - we neither assert on the result nor assign to `_`, right?

Comment (sunshowers) [reply nexus/reconfigurator/execution/src/test_utils.rs]: Done -- hope that's clearer!

Comment (jgallagher) [reply nexus/reconfigurator/execution/src/test_utils.rs]: It does - thanks!

**Review by jgallagher — APPROVED** [review]

**Pull request merged**